bokehjs/src/lib/core/util/arrayable.ts

```diff
@@ -201,3 +201,52 @@ export function sorted_index<T>(array: Arrayable<T>, value: T): number {
   }
   return low
 }
+
+export function bin_counts(data : Arrayable<number>, bin_edges : Arrayable<number>) {
+    const counts = Array(bin_edges.length -1).fill(0)
+    for (let i = 0; i < data.length; i++){
+        const sample = data[i]
+        for (let bin = 0; bin < counts.length; bin++){
+            if(sample > bin_edges[bin] && sample <= bin_edges[bin+1]){
+                counts[bin]+=1
+            }
+        }
+    }
+    return counts
+}
+
+export function interp (pointsToEvaluate: number[], functionValuesX: Arrayable<number>, functionValuesY: Arrayable<number>) : number[] {
+    const results : number[] = [] 
+    pointsToEvaluate.forEach(function (point: number) {
+    let index = leftedgeindex(point, functionValuesX)
+    if (index == functionValuesX.length - 1)
+      index--
+    results.push(lerp(point, functionValuesX[index], functionValuesY[index]
+      , functionValuesX[index + 1], functionValuesY[index + 1]))
+  })
+  return results
+}
+
+function lerp (x :number, x0:number, y0 :number,
+                              x1:number, y1:number) :number {
+  const a = (y1 - y0) / (x1 - x0)
+  const b = -a * x0 + y0
+  return a * x + b
+}
+
+function leftedgeindex(point : number, intervals: Arrayable<number>) :number {
+  if (point < intervals[0])
+    return 0
+  if (point > intervals[intervals.length - 1])
+    return intervals.length - 1
+  let indexOfNumberToCompare 
+    , leftEdgeIndex = 0
+    , rightEdgeIndex = intervals.length - 1
+  while (rightEdgeIndex - leftEdgeIndex !== 1) {
+    indexOfNumberToCompare = leftEdgeIndex + Math.floor((rightEdgeIndex - leftEdgeIndex)/2)
+    point >= intervals[indexOfNumberToCompare]
+      ? leftEdgeIndex = indexOfNumberToCompare
+      : rightEdgeIndex = indexOfNumberToCompare
+  }
+  return leftEdgeIndex
+}
```

Comment:
A few comments: 

* "lerp" is a terrible name :) We should be explicit e.g `linear_interpolate` and also there is at least one other place that does linear interpolation and we should share code between: https://github.com/bokeh/bokeh/blob/master/bokehjs/src/lib/models/transforms/linear_interpolator.ts#L47

* I don't think this is the right module for these. Maybe the math module, maybe a new util module, open to suggestion

* Simple functions like these should get tests